### PR TITLE
Issue 130/search document styling updates

### LINKED ISSUE #130: Search Document Styling Updates
Adding MUI styling to the Search Document component based on designs from the UX Figma board.

Currently prioritizing this over Upload Document Styling Updates #117 in order to build off of smaller components into more complex ones.

### PR DESCRIPTION
**Pull request by andycwilliams**
This branch is unfinished. Feedback needed.

You'll notice that I also changed how the styling for all forms works. It removes the `.panel` styling from the CSS file. It's pretty light but more consistent with the new MUI theming (with the exception of the `Logout` button, which will likely be moved to the menu anyway). The rest of the CSS file styling still needs to be transferred over to `theme.js`.

The `Get Document` button should work properly.

Switching to the new select option, however, is not as smooth. You can see that there is the previous select option as well as the new MUI version. The MUI version is not yet connected to the actual functionality and so is right now acting as a demonstration.

Before continuing, I wanted to know how we want to approach this. Do we want to rework the way `DocumentSelection.jsx` (and thus `docTypes.js`) works? For instance, MUI has its own method of iterating through options. 

By developing a plan here, styling the remaining components will be simpler.

### CODE REVIEW

**Comment by xscottxbrownx** [comment]
> This branch is unfinished. Feedback needed.
> 
> You'll notice that I also changed how the styling for all forms works. It removes the `.panel` styling from the CSS file. It's pretty light but more consistent with the new MUI theming (with the exception of the `Logout` button, which will likely be moved to the menu anyway). The rest of the CSS file styling still needs to be transferred over to `theme.js`.
> 
> The `Get Document` button should work properly.
> 
> Switching to the new select option, however, is not as smooth. You can see that there is the previous select option as well as the new MUI version. The MUI version is not yet connected to the actual functionality and so is right now acting as a demonstration.
> 
> Before continuing, I wanted to know how we want to approach this. Do we want to rework the way `DocumentSelection.jsx` (and thus `docTypes.js`) works? For instance, MUI has its own method of iterating through options.
> 
> By developing a plan here, styling the remaining components will be simpler.

@leekahung if you can weigh in here too, I believe Andy is asking about the functional logic here.

---

@andycwilliams Honestly, I'm a little confused and not sure if it would be easier to work out/explain over a Voice Chat. 

---

**But, is there a `docTypes.js`?**

I see, I found it inside `form-helper.js` 
>That already is strange to me. It's just an array of options and set as a constant. Wouldn't this just belong in a `constants` >folder? And naming convention be `DOCTYPES` or `DOC_TYPES`?  

Nevertheless, since your `handleChange` function sets the document type state, could we just make `const docTypes = documentType`?  Line 41 here:
Screenshot 2023-05-16 at 2 06 27 PM


---

Seems to me all `<DocumentSelection>` component does is map over the array of possible document types (`docTypes`) and displays them as options. This is probably helpful re-usable code for all the forms. 
If everybody is in agreement on that, I believe that should be styled with MUI (basically your `<Select>` and `<MenuItem>` elements.)

**So, `FetchDocument.jsx` would be something like this:**
Screenshot 2023-05-16 at 2 53 32 PM

**And `DocumentSelection.jsx` would be something like this:**
Screenshot 2023-05-16 at 2 53 48 PM

**which results in:**

https://github.com/codeforpdx/PASS/assets/71395931/4ec8ad28-ba67-479d-9153-54577a054afc

---

Screenshot 2023-05-16 at 2 37 13 PM

Personally, I don't think we need this label - the placeholder does the job, could change that to "Select Document Type" if you wish.

**Comment by leekahung** [comment]
I think where you and Scott are heading towards is pretty good for the styling.

Ideally, we'll want to make that form as simple as possible since the `FetchDocument.jsx` form is literally just a dropdown selector with a button and a notification component attached to it.

The notification component would probably have to be refactored as well, but we can save that for later.

As for `DocumentSelection.jsx`, we should refactor that as well that with MUI components as shown by Scott.

**Comment by andycwilliams** [comment]
Thanks for the suggestions. It seems to be working just fine for me.

If it looks good, I can move on to another smaller component, such as Delete Document.

Actually, there might be an issue with fetching the document types. I'll take another look tomorrow.

File: src/components/Form/FormSection.jsx
Comment (leekahung) [inline src/components/Form/FormSection.jsx:1]: Probably related to FormSection or Footer, there seem to be no gap between them in the Home page and the Forms page. (see example below)

Screen Shot 2023-05-17 at 12 08 32

Would be good if we keep it consistent like what we have now with the Navbar and FormSection.

Screen Shot 2023-05-17 at 12 10 57

File: src/utils/form-helper.js
Comment (leekahung) [inline src/utils/form-helper.js]: Makes sense to change it to "Driver's License". But we'll need to update the getContainerUrl in session-helper.

I think we can be done easily by changing Line 169 in session-helper.js which uses

```
case 'Drivers License':
```

to

```
case "Driver's License":
```

This will help us select the correct URL path for the container.

File: src/components/Form/DocumentSelection.jsx
Comment (leekahung) [inline src/components/Form/DocumentSelection.jsx:1]: Since we're dealing with a component what's being used specifically for any forms with select, we should go and change all the HTML `<form>` elements with `<DocumentSelection>` to MUI `<FormControl>`components like what is done with FetchDocumentForm.jsx

Else it'll break everything:

Screen Shot 2023-05-17 at 12 13 10

https://github.com/codeforpdx/PASS/assets/14917816/c3db8574-9a71-4c47-b4e5-6f60ab2dad36

As stated in the error message, it's because the current component has uncontrolled inputs. This should clear up  once all the forms with DocumentSelection is converted to using FormControl.

**Review by leekahung — CHANGES_REQUESTED** [review]
See feedback for requested changes.

File: src/components/Form/DocumentSelection.jsx
Comment (andycwilliams) [reply src/components/Form/DocumentSelection.jsx:1]: This is marked as Outdated. Do you not want to pursue this? It's the one thing I haven't changed yet in my most recent commit. Edit: Or does Outdated just mean the file in question has been updated since the comment was made?

When I tried transferring from `<form>` to MUI `<FormControl>`, it would work. So I'd need to take a closer look at it.

As is, it seems to be working. At least for this one component.

Comment (leekahung) [reply src/components/Form/DocumentSelection.jsx:1]: Oh, for the "Outdated" marker on GitHub if you made any kind of new commits after the initial comment for that specific file.

Comment (leekahung) [reply src/components/Form/DocumentSelection.jsx:1]: > When I tried transferring from `<form>` to MUI `<FormControl>`, it would work. So I'd need to take a closer look at it.
> 
> As is, it seems to be working. At least for this one component.

With regards to this however, I would prefer getting this updated `<form>` to `<FormControl>` for any form components with DocumentSelection.jsx if you decide to update this. It just breaks too many things without the migration.

You could simply use what's being done in FetchDocumentForm.jsx as a guide.

Comment (andycwilliams) [reply src/components/Form/DocumentSelection.jsx:1]: Just so I understand, is this what you mean? I don't have a ton more time to work on it today and do more tomorrow.



Also I wanted to talk about `style={formRowStyle}`. It looks like just adding some margin and is present on a lot of the forms. Seems like it could be consolidated in `theme.js`, `FormSection.jsx`, or simply inline on each component such as in the existing `sx` property on line 80.

Not a major issue, but could save a couple extra bits of code.

Comment (leekahung) [reply src/components/Form/DocumentSelection.jsx:1]: Yeah, what is shown in the image is when I meant.

If we're updating DocumentSection to Selection/MenuItem, I think what needs to be done is to make the form have controlled components by MUI standards. This meant to have `<form>` be replaced with `<FormControl>` and `<label>` with `<InputLabel>` at least with regards to forms with DocumentSelection.

I've tried running some of the functions with the error, I've shown previously, but it gives me a different error with event.target.<property>.value. Having it fixed should stop the error messages from appear and will allow React to pick up the target value properly.

Comment (leekahung) [reply src/components/Form/DocumentSelection.jsx:1]: As for the style attribute, feel free to phase that out.

Comment (andycwilliams) [reply src/components/Form/DocumentSelection.jsx:1]: Okay, today is pretty packed so I can work on that and should be ready tomorrow. I think that would then be enough to close the PR?

Comment (leekahung) [reply src/components/Form/DocumentSelection.jsx:1]: I think so. We're just missing the corrections for these uncontrolled forms at the moment if the other requested changes are fixed (like the gap/margin problem with the Footer and the last form component of the page)

Screen Shot 2023-05-17 at 12 08 32

With both these fixes I think it'll be ready to merge.

**Comment by andycwilliams** [comment]
Still in progress.

The footer gap issue should be resolved.

Since multiple forms will have to be updated, rather than just one component, I want to make sure I'm on the right track. 

I want to see if `Search Document` is working for others now. It seems to be fine for me. I get a 404 error for one type of file, but it is also the type that is missing from my Pod. So I believe that is why.

I've made similar changes to `Delete Document` and that seems functional as well.

If these look good to others, I can finish applying the document selection update to all the other relevant forms.

**Comment by leekahung** [comment]
> If these look good to others, I can finish applying the document selection update to all the other relevant forms.

Looks like you're on the right track. Feel free to finishing applying those changes.

**Comment by andycwilliams** [comment]
I've made some more changes but there's still more to do. I'm concerned about the scope of this PR expanding too much.

By changing the search document function in each form, more has to be changed in order to get it to work. So it seems like this PR will just have to tackle more than originally intended since not every form is working again yet.

**Comment by leekahung** [comment]
I don't think we're going out of scope considering DocumentSelection.jsx itself covers multiple form components. If we're going to change DocumentSelection.jsx, components uses it should be included in the updates.

**Comment by leekahung** [comment]
Not for this PR, but I think we can definitely touch out the layout of individual forms later.

Screen Shot 2023-05-18 at 16 09 27

File: src/components/Form/UploadDocumentForm.jsx
Comment (leekahung) [inline src/components/Form/UploadDocumentForm.jsx]: The FormControl in Line 142 are not implemented correctly. `<FormControl>` is suppose to replace `<form>` in Line 140 not the `<label>` element (we have `<InputLabel>` for that).

See how example in FetchDocumenForm.jsx where the `<form>` element is completely replaced by `<FormControl>` and where `<input>` is replaced with `<InputLabel>`

File: src/components/Form/CheckAclPermsDocContainerForm.jsx
Comment (leekahung) [inline src/components/Form/CheckAclPermsDocContainerForm.jsx]: Why is useState imported here? I'm not seeing it being used in this form.

File: src/components/Form/CheckAclPermsDocContainerForm.jsx
Comment (leekahung) [inline src/components/Form/CheckAclPermsDocContainerForm.jsx]: Until they're used in this form, I don't think we need to import any MUI components here.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodQueryForm.jsx]: We already have state set up for username (see Line 29), we do not need another useState for filling out that form.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodQueryForm.jsx]: Related to Line 32, we already have a state that grabs our username. Line 29 also contains the onChange function within {...username}. Revert this back to event.target.crossPodQuery.value.

The only reason why we're using docType like that is because we've refactored how the state is called with the new Select component in DocumentSelection.jsx.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodQueryForm.jsx]: Related to Line 32, we already have a state that grabs our username. Line 29 also contains the onChange function within {...username}. Revert Line 112 to 115 back to {...username}.

File: src/components/Form/CrossPodQueryForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodQueryForm.jsx]: `<FormControl>` is a `<form>` element, not a `<div>` or a `<label>`. Use `<FormControl>` to replace the `<form>`. This means Lines 102 to 118 should be inside `<FormControl>`

File: src/components/Form/CrossPodWriteForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodWriteForm.jsx]: Same as last comment,`<FormControl>` is a `<form>` element. It's not a `<div>` or a `<label>`. Use `<FormControl>` to replace the `<form>`. This means Lines 166 to 177 and Lines 186 to 204 should be inside `<FormControl>`

File: src/components/Form/CrossPodWriteForm.jsx
Comment (leekahung) [inline src/components/Form/CrossPodWriteForm.jsx]: This button should be outside of `<FormControl>` following FetchDocumentsForm.jsx

File: src/components/Form/ManageUsers.jsx
Comment (leekahung) [inline src/components/Form/ManageUsers.jsx]: Only import components that are being used. If they're not used yet, leave them out.

File: src/components/Form/SetAclPermissionForm.jsx
Comment (leekahung) [inline src/components/Form/SetAclPermissionForm.jsx]: `<FormControl>` is a `<form>` element, not a `<div>` or a `<label>`. Use `<FormControl>` to replace the `<form>`. That means Line 119 to 129 and Line 141 to 147 should be inside `<FormControl>`.

**Review by leekahung — CHANGES_REQUESTED** [review]
I'm a bit confused as to the recent commits. We are getting MUI components integrate, but not in the correct way. See feedback.

**Comment by xscottxbrownx** [comment]
Here's my comment/thought to make this easier and keep the scope smaller.
---
I would say to just do ONLY the required changes in all the form components to make sure they work without warnings/errors.  **Meaning only change these elements in those files:**

`<form>` = `<FormControl>`
`<label>` = `<InputLabel>`
`<input>` = correct MUI input component.  If it is a select component and the options to select should be "Bank Statement", "Passport", and  "Driver's License" then use our custom `<DocumentSelect>`

---

**Get these working, and we can merge this PR.**  
Then go back to the small separate PR's of implementing MUI completely on each form component one at a time.

**Comment by leekahung** [comment]
> ## Here's my comment/thought to make this easier and keep the scope smaller.
> I would say to just do ONLY the required changes in all the form components to make sure they work without warnings/errors. **Meaning only change these elements in those files:**
> 
> `<form>` = `<FormControl>` `<label>` = `<InputLabel>` `<input>` = correct MUI input component. If it is a select component and the options to select should be "Bank Statement", "Passport", and "Driver's License" then use our custom `<DocumentSelect>`
> 
> **Get these working, and we can merge this PR.** Then go back to the small separate PR's of implementing MUI completely on each form component one at a time.

Thanks Scott. Yeah, this is what I had in mind for `<FormControl>` and `<InputLabel>` and `<DocumentSelect>`. The changes should be minor to go from `<form>` to `<FormControl>`, `<label>` to `<InputLabel>`, etc.

File: src/components/Form/CheckAclPermsDocContainerForm.jsx
Comment (andycwilliams) [reply src/components/Form/CheckAclPermsDocContainerForm.jsx]: I was getting overly ambitious with my restyling of components and when I dialed things but I left in some imports. I expect to end up needing them in the future, but will remove them now.

**Comment by andycwilliams** [comment]
Alright, so this still isn't finished (I haven't applied MUI styling to each part of each component (e.g. CrossPodWriteForm). Also, I need to test functionality more and ensure that nothing is broken.

I will continue working on these but first I want to make sure I address the current issues.

EDIT: Just pulled from the latest Development update and am working on resolving conflicts. Will push a new commit soon.

**Comment by andycwilliams** [comment]
As before, this is unfinished. 

I've merged with Development and resolved the conflicts, updating quite a few files in the process.

More styling is applied (such as previously unimplemented MUI components like checkboxes and calendars) but all the functionality still needs testing.

With this latest commit, I am looking for feedback on the progress of the styling and whether it has addressed the previous change recommendations. Then I can continue working on it.

**Comment by leekahung** [comment]
> With this latest commit, I am looking for feedback on the progress of the styling and whether it has addressed the previous change recommendations. Then I can continue working on it.

While style-wise it looks alright, the issues to the underlying codebase and functionality has not been addressed unfortunately. The commits seems to be getting more complicated than it needs to be rather than heading towards a solution. Ideally, we're just changing out HTML elements with MUI components. They're not exactly one to one because of how MUI name their components and how MUI uses it compared to HTML, but if we have something in the line of:

```HTML
<form>
  <label htmlFor="something">This label<label>
  <input id="something"/>
</form>
```

The MUI equivalent should be:

```JSX
<FormControl>
  <InputLabel htmlFor="something">This label</InputLabel>
  <Input id="something" />
</FormControl>
```

This is what is shown in MUI's documentation for `<FormControl>`: https://mui.com/material-ui/api/form-control/

The same can be said for `<select>` in HTML, that's just `<Select>` in MUI and replacing `<input>` with `<select>` is exactly the same as you would for HTML, but you're just changing `<option>` with `<MenuItem>` (see https://mui.com/material-ui/react-select/):

MUI:

```JSX
<FormControl>
  <InputLabel id="demo-simple-select-label">Age</InputLabel>
  <Select
    labelId="demo-simple-select-label"
    id="demo-simple-select"
    value={age}
    label="Age"
    onChange={handleChange}
  >
    <MenuItem value={10}>Ten</MenuItem>
    <MenuItem value={20}>Twenty</MenuItem>
    <MenuItem value={30}>Thirty</MenuItem>
  </Select>
</FormControl>
```

**Comment by xscottxbrownx** [comment]
> The same can be said for `<select>` in HTML, that's just `<Select>` in MUI and replacing `<input>` with `<select>` is exactly the same as you would for HTML, but you're just changing `<option>` with `<MenuItem>` (see https://mui.com/material-ui/react-select/):
> 
> MUI:
> 
> ```js
> <FormControl>
>   <InputLabel id="demo-simple-select-label">Age</InputLabel>
>   <Select
>     labelId="demo-simple-select-label"
>     id="demo-simple-select"
>     value={age}
>     label="Age"
>     onChange={handleChange}
>   >
>     <MenuItem value={10}>Ten</MenuItem>
>     <MenuItem value={20}>Twenty</MenuItem>
>     <MenuItem value={30}>Thirty</MenuItem>
>   </Select>
> </FormControl>
> ```

**Unless** that `<Select>` is combined with `<MenuItem>`s of the 3 document types we have, then just use our custom `<DocumentSelection>` component instead of `<Select>`

File: src/components/Form/FetchDocumentForm.jsx
Comment (leekahung) [inline src/components/Form/FetchDocumentForm.jsx]: I think it's safe enough to turn these off now that our labels are now properly control with FormControl. I've turned removed the linters for this page and it seems to not show any errors even after running the linters manually with npm run lint.

File: src/components/Form/FetchDocumentForm.jsx
Comment (leekahung) [inline src/components/Form/FetchDocumentForm.jsx]: Same as previous disable lint line. I think we can safely remove this.

**Review by leekahung — COMMENTED** [review]
Of course, as you mentioned to me, it might still need styling updates. However, the code looks pretty clean now.

Things are still working as intended in the FetchDocumentForm, it shouldn't be far from a full transition to MUI.

Mainly just comments with regards to the linters, otherwise, great work!

**Comment by leekahung** [comment]
Looks good to me now.

Think it's ready for merging. But for now, I'll have this on hold until Development is merged to Master.

**Review by leekahung — APPROVED** [review]
This looks fine. I approve.

**Pull request merged**
